src/OpenTelemetry.Instrumentation.Runtime/RuntimeMetrics.cs

```diff
@@ -113,14 +123,27 @@ private static double GetFragmentation()
         }
 #endif
 
-        private IEnumerable<Measurement<long>> GetProcessorTimes()
+#if NET6_0_OR_GREATER
+        private static IEnumerable<Measurement<long>> GetGarbageCollectionHeapSizes()
         {
-            var process = Process.GetCurrentProcess();
-            return new[]
+            var generationInfo = GC.GetGCMemoryInfo().GenerationInfo;
+
+            Measurement<long>[] measurements = new Measurement<long>[generationInfo.Length];
```

Comment:
This will cause allocation. We could either use a pre allocated array or make use of the `yield return` semantic.

Reply:
This metrics is an Observable and gets called once every 15~20 seconds. So it's not worth doing too much optimization for now. Most importantly, if there are multiple readers reading it at almost the same time, they could have incorrect values.

I've tried to use `yield return` and it worked for two other methods, however, for this one I got an error. See discussion thread at https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/412#discussion_r894070227

> The compile will fail with GetGarbageCollectionHeapSizes because it needs a `Span<T>`: "CS4013 Instance of type 'ReadOnlySpan' cannot be used inside a nested function, query expression, iterator block or async method".